### PR TITLE
DEN-19 Access rule Read

### PR DESCRIPTION
**Pull request by darshandevexp**

### CODE REVIEW

File: ec/resources/waf/access_rule.go [hunk 1]
```diff
@@ -458,11 +458,58 @@ func ResourceAccessRuleCreate(ctx context.Context, d *schema.ResourceData, m int
...
+	(*config).AccountNumber = accountNumber
+
+	wafService, err := buildWAFService(**config)
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	//change the order
+	log.Printf("[INFO] Reading WAF Access Rule Id %s for Account >> %s", accountNumber, ruleID)
```
Comment (stevenpaz) [inline ec/resources/waf/access_rule.go:478]: Looks like you still need to change the order here :)

File: ec/resources/waf/access_rule.go [hunk 2]
```diff
@@ -458,11 +458,58 @@ func ResourceAccessRuleCreate(ctx context.Context, d *schema.ResourceData, m int
...
+
+	resp, err := wafService.GetAccessRuleByID(accountNumber, ruleID)
+
+	if err != nil {
+		d.SetId("")
+		return diag.FromErr(err)
+	}
+
+	//change %v with printing funcation.
+	log.Printf("[INFO] Retrieved Rule: %+v", resp)
```
Comment (stevenpaz) [inline ec/resources/waf/access_rule.go:488]: Don't forget to implement this printing function

File: ec/resources/waf/access_rule_test.go [hunk 3]
```diff
@@ -85,3 +85,63 @@ func TestConvertInterfaceToAccessControls(t *testing.T) {
 		}
 	}
 }
+func TestFlattenAccessControls(t *testing.T) {
```
Comment (stevenpaz) [inline ec/resources/waf/access_rule_test.go:88]: Add a space between functions

File: ec/resources/waf/access_rule_test.go [hunk 4]
```diff
@@ -85,3 +85,63 @@ func TestConvertInterfaceToAccessControls(t *testing.T) {
...
+					"blacklist": []string{
+						"10.10.10.3",
+					},
+					"whitelist": []string{
+						"10.10.10.2",
+					},
+				},
+			},
+		},
+	}
```
Comment (stevenpaz) [inline ec/resources/waf/access_rule_test.go:116]: Can you add one or two edge cases?

**Review by stevenpaz — CHANGES_REQUESTED** [review]

File: ec/resources/waf/access_rule_test.go (hunk 3 above)
Comment (darshandevexp) [reply ec/resources/waf/access_rule_test.go:88]: so spacing is being removed go formatter but just to confirm we should have function name func TestFlattenAccessControls (t *testing.T) {

Comment (stevenpaz) [reply ec/resources/waf/access_rule_test.go:88]: Hmm ok

**Comment by darshandevexp** [comment]
@stevenpaz ready for review

**Review by stevenpaz — APPROVED** [review]

**Pull request merged**